lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -2459,7 +2459,7 @@ namespace Js
             Var member;
 
             // If the item is found in the array own body, then it is a number
-            if (JavascriptOperators::GetOwnItem(object, index, &member, scriptContext))
+            if (JavascriptOperators::GetOwnItem(object, index, &member, scriptContext) && member != scriptContext->GetLibrary()->GetUndefined())
```

Comment:
Can you move this check to the separate next line? It is bit confusing for me.   
This is safe fix for 1.2  If GetItem returns false for undefined, other code paths might walk the prototype chain, which is not desirable for typedarray.